rust/agama-lib/src/software/settings.rs

```diff
@@ -42,6 +44,45 @@ pub struct SoftwareSettings {
     pub only_required: Option<bool>,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(untagged)]
+pub enum PatternsSettings {
+    PatternsList(Vec<String>),
```

Comment:
Why not just `List`? `PatternsSettings::PatternsList` sounds sort of repetitive.